Userland/Services/Taskbar/main.cpp

```diff
@@ -176,42 +176,26 @@ ErrorOr<NonnullRefPtr<GUI::Menu>> build_system_menu(GUI::Window& window)
         }
 
         auto parent_menu = app_category_menus.get(app.category).value_or(system_menu.ptr());
-        parent_menu->add_action(GUI::Action::create(app.name, icon, [app_identifier](auto&) {
+        parent_menu->add_action(GUI::Action::create(app.name, icon, [app_identifier, &window](auto&) {
             dbgln("Activated app with ID {}", app_identifier);
             auto& app = g_apps[app_identifier];
-            char const* argv[4] { nullptr, nullptr, nullptr, nullptr };
+            StringView executable;
+            Vector<char const*, 2> arguments;
             auto pls_with_executable = DeprecatedString::formatted("/bin/pls {}", app.executable);
```

Comment:
I'm just noticing that this will break for executables that contain spaces, they should be surrounded with single quotes and the executable name's quotes escaped. Can you add a FIXME?

Reply:
Easy to just do the obvious fix at the same time (won't handle the edge case of paths that contain single quotes, so I've left a FIXME for that). However, I don't think this functionally is actually use at all, so maybe we could just remove it? I don't think we have any terminal apps, that require root, that you can run from the Serenity menu.